### PR TITLE
Avoid auto complete for OpenID scopes

### LINKED ISSUE #13184: Don't let browsers auto-fill Scopes and Client Secret OpenId fields
As you can see in the screenshot, scopes and client secret are always filled in automatically,
And the content of the padding doesn't have any relevance
This might be a Chromium issue, but we should be able to turn it off by disabling the form's autofill property
I tried setting 'autocomplete='off'' on the input a few months ago and it didn't seem to work

image

### PR DESCRIPTION
**Pull request by hishamco**
Fixes #13184

### CODE REVIEW

**Comment by hishamco** [comment]
@hyzx86 in which browser you saw the bug?

**Comment by hyzx86** [comment]
Edge, Maybe this problem has been fixed. I'll check it later.

________________________________

File: src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml [hunk 1]
```diff
@@ -54,7 +54,7 @@
 
 <div class="mb-3" asp-validation-class-for="Scopes">
     <label asp-for="Scopes" class="form-label">@T["Scopes"]</label>
-    <input asp-for="Scopes" class="form-control" />
+    <input asp-for="Scopes" class="form-control"autocomplete="off" />
```
Comment (hyzx86) [inline src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml:57]: Referring to the screenshot below, there are two places where it is recommended to add directly to the form element

Also, the added attribute omits the space ：）
image

**Review by hyzx86 — CHANGES_REQUESTED** [review]

File: src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml (hunk 1 above)
Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml:57]: > there are two places where it is recommended to add directly to the form element

Where are they?

> Also, the added attribute lost a space ：）

Really? which browser? coz we using this attribute in several places

Comment (hyzx86) [inline src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml:57]:
```suggestion
    <input asp-for="Scopes" class="form-control" autocomplete="off" />
```

Comment (hyzx86) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml:57]: Ignore the screenshot above, it appears to be out of date

I'm using Edge, the other place I'm referring to is `Client Secret`, the latest version of the code I looked at seems to have added the autocomplete property.

Comment (hyzx86) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml:57]: OK , the above screenshot is indeed from the latest version of the source code, the Scopes element has indeed not been adjusted, but the `Client Secret` has been adjusted, and it doesn't seem to be working as expected, it's still autofilling (note that the textboxes in the red area on the left contain a light blue background).

image

**Review by MikeAlhayek — APPROVED** [review]

File: src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml (hunk 1 above)
Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml]: Do you mean the current auto-filling? I don't think so. Please the tested browser

Comment (Piedone) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml]: The Client Secret needs to be not auto-complete too indeed. I see the same issue with Chrome.

Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml]: The auto-complete is already Off, if you tested recently I will check in Chrome

Comment (Piedone) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml]: If you have credentials saved for the host, then it'll auto-fill them here still:

image

I have admin/Password1! saved for OC login. Check it after saving something like this.

Comment (hyzx86) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml]: This is a strange problem, I applied the changes in the code, but the problem still exists, if you open and save in an anonymous window, the autocompletion really doesn't work, but it still autofills in the default browser!

Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml]: I didn't see the Client Secret in my screen, is there any configuration

Comment (hyzx86) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml]: > I didn't see the Client Secret in my screen, is there any configuration

Yes， you have to check  `Use code response type` , Refer to the screenshot above

**Comment by MikeAlhayek** [comment]
@hishamco what is the status of this PR? This is a good PR that is fixing an issue.

**Review by Piedone — APPROVED** [review]

**Comment by Piedone** [comment]
Anybody with anything else or can we merge?

File: src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml [hunk 2]
```diff
@@ -144,7 +144,7 @@
 
 <div class="mb-3 collapse" asp-validation-class-for="ClientSecret">
     <label asp-for="ClientSecret" class="form-label">@T["Client Secret"]</label>
-    <input asp-for="ClientSecret" class="form-control" type="password" autocomplete="off" />
+    <input asp-for="ClientSecret" class="form-control" type="password" autocomplete="new-password" />
```
Comment (hyzx86) [inline src/OrchardCore.Modules/OrchardCore.OpenId/Views/OpenIdClientSettings.Edit.cshtml:147]: This should be a good solution 👍

**Review by hyzx86 — APPROVED** [review]

**Pull request merged**